src/main/java/org/checkerframework/specimin/JavaParserUtil.java

```diff
@@ -68,4 +73,64 @@ public static ResolvedReferenceType classOrInterfaceTypeToResolvedReferenceType(
       ClassOrInterfaceType type) {
     return type.resolve().asReferenceType();
   }
+
+    /**
+     * Searches the ancestors of the given node until it finds a class or interface node, and then
+     * returns the fully-qualified name of that class or interface.
+     *
+     * <p>This method will fail if it is called on a node that is not contained in a class or
+     * interface.
+     *
+     * @param node a node contained in a class or interface
+     * @return the fully-qualified name of the inner-most containing class or interface
+     */
+    @SuppressWarnings("signature") // result is a fully-qualified name or else this throws
+    public static @FullyQualifiedName String getEnclosingClassName(Node node) {
+      Node parent = getEnclosingClassLike(node);
+
+      if (parent instanceof ClassOrInterfaceDeclaration) {
+        return ((ClassOrInterfaceDeclaration) parent).getFullyQualifiedName().orElseThrow();
+      }
+
+      if (parent instanceof EnumDeclaration) {
+        return ((EnumDeclaration) parent).getFullyQualifiedName().orElseThrow();
+      }
+
+      throw new RuntimeException("unexpected kind of node: " + parent.getClass());
+    }
+
+    /**
+     * Returns the innermost enclosing class-like element for the given node. A class-like element is
+     * a class, interface, or enum (i.e., something that would be a {@link
+     * javax.lang.model.element.TypeElement} in javac's internal model). This method will throw if no
+     * such element exists.
+     *
+     * @param node a node that is contained in a class-like structure
+     * @return the nearest enclosing class-like node
+     */
+    public static Node getEnclosingClassLike(Node node) {
+      Node parent = node.getParentNode().orElseThrow();
+      while (!(parent instanceof ClassOrInterfaceDeclaration || parent instanceof EnumDeclaration)) {
```

Comment:
This method should also handle `AnnotationDeclaration` since annotations are considered an interface under `javax.lang.model.element.TypeElement`. This change is made in #302, (in `PrunerVisitor`, however), so we could make the change in that PR instead of here.